scarlet/lite/component.py

```diff
@@ -0,0 +1,621 @@
+    """Gradient of the the component model wrt the Gaussian morphology parameters
+
+    Parameters
+    ----------
+    input_grad: `numpy.ndarray`
+        Gradient of the likelihood wrt the component model
+    params: `numpy.ndarray`
+        The parameters of the morphology.
+    cls: `LiteComponent`
+        The component of the model that contains the morphology.
+    morph: `numpy.ndarray`
+        The model of the morphology.
+    sed: `numpy.ndarray`
+        The model of the SED.
+    ellipse: `EllipseFrame`
+        The ellipse parameters to scale the radius in all directions.
+    """
+    n = params[5]
+    bn = 2 * n - 1 / 3 + 4 / 405 / n + 46 / 25515 / n ** 2 + 131 / 1148175 / n ** 3 + 2194697 / 30690717750 / n ** 4
+    if n == 1:
+        # Use a simplified model for faster calculation
+        dExp = -bn * morph
+    else:
+        r = ellipse.R
+        dExp = -bn / n * morph * r ** (1 / n - 1)
+
+    _grad = np.zeros(cls.bbox.shape, dtype=morph.dtype)
+    _grad[cls.slices[1]] = input_grad[cls.slices[0]]
+    _grad = np.einsum("i,i...", sed, _grad)
+    dN = np.sum(_grad * bn * morph * ellipse.R**(1/n) * np.log10(ellipse.R)/n**2)
+    _grad = _grad * dExp
+    dY0 = ellipse.grad_y0(_grad, False)
+    dX0 = ellipse.grad_x0(_grad, False)
+    dSigmaY = ellipse.grad_major(_grad, False)
+    dSigmaX = ellipse.grad_minor(_grad, False)
+    dTheta = ellipse.grad_theta(_grad, False)
+    return np.array([dY0, dX0, dSigmaY, dSigmaX, dTheta, dN], dtype=params.dtype)
+
+
+class ParametricComponent:
+    """A parametric model
+    """
+
+    def __init__(self, sed, morph_params, morph_func, morph_grad, morph_prox,
+                 morph_step, model_frame, bbox, prox_sed=None, floor=1e-20):
+        """Initialize the component
+
+        Parameters
+        ----------
+        sed: `numpy.ndarray`
+            The SED of the component.
+        morph_params: `numpy.ndarray`
+            The parameters of the morphology.
+        morph_func: `Function`
+            The function to generate the morpholgy
+            based on `morphParams`.
+        morph_grad: `Function`
+            The function to calculate the gradient of the
+            likelihood wrt the morphological parameters.
+        morph_prox: `Function`
+            The proxmial operator for the morphology parameters.
+        bbox: `scarlet.bbox.Box`
+            The bounding box that holds the model.
+        frame: `CartesianGrid`
+            The coordinates of the model frame,
+            used to speed up the creation of the
+            polar grid for each source.
+        prox_sed: `Function`
+            Proximal operator for the SED.
+            If `prox_sed` is `None` then the default proximal
+            operator `self.prox_sed` is used.
+        floor: `float`
+            The minimum value of the SED, used to prevent
+            divergences in the gradients.
+        """
+        params = FixedParameter(morph_params)
+        sed = FixedParameter(sed)
```

Comment:
Yes, it's a single array, where the array is interpreted as a parameter (as is custom in automatic gradient packages). I get your point and see how it can be confusing, but I think that it would be more confusing to use a different nomenclature than the standard usage.